src/configure/templateInputHelper/InputControl.ts

```diff
@@ -110,11 +130,41 @@ export class InputControl {
 
         if (!!properties) {
             value = properties[propertyName];
-            if( !!value && !!inputs ) { return MustacheHelper.render(value, { inputs: inputs }); }
+            if (!!value && !!inputs) { return MustacheHelper.render(value, { inputs: inputs }); }
         }
         return value;
     }
 
+    public setValidationDataSources(dataSourceToInputsMap: Map<DataSource, InputControl[]>) {
+        this.validationDataSourceToInputsMap = dataSourceToInputsMap;
+    }
+
+    public async triggerControlValueValidations(value: string): Promise<string> {
+        this._tempValue = value;
```

Comment:
temp is an ambiguous word. If I understand correctly it is the value which has been selected by user, has not yet been validated. If so, should we call it _valuePendingValidation  or something similar?

Reply:
yes, changed it.